.github/workflows/tool-crucible-ci.yaml

```diff
@@ -0,0 +1,148 @@
+  cancel-in-progress: true
+
+env:
+  GITHUB_WORKSPACE: ${{ inputs.github_workspace }}
+
+jobs:
+  gen-params:
+    runs-on: ubuntu-latest
+    outputs:
+      userenvs: ${{ steps.get-userenvs.outputs.userenvs }}
+      github_hosted_scenarios: ${{ steps.get-scenarios-github.outputs.scenarios }}
+      repo_name: ${{ steps.get-repo-name.outputs.repo-name }}
+    steps:
+
+    - name: checkout crucible-ci default
+      if: ${{ inputs.crucible_ci_test != 'yes' }}
+      uses: actions/checkout@v3
+      with:
+        repository: perftool-incubator/crucible-ci
+        ref: main
+        path: crucible-ci
+    - name: checkout crucible-ci crucible_ci_test
+      if: ${{ inputs.crucible_ci_test == 'yes' }}
+      uses: actions/checkout@v3
+      with:
+        repository: perftool-incubator/crucible-ci
+        ref: ${{ inputs.crucible_ci_test_branch }}
+        path: crucible-ci
+
+    - name: checkout rickshaw
+      uses: actions/checkout@v3
+      with:
+        repository: perftool-incubator/rickshaw
+        ref: master
+        path: rickshaw
+    - name: run get-userenvs
+      id: get-userenvs
+      uses: ./crucible-ci/.github/actions/get-userenvs
+      with:
+        rickshaw-directory: "./rickshaw"
+    - name: run get-scenarios-github
+      id: get-scenarios-github
+      uses: ./crucible-ci/.github/actions/get-scenarios
+      with:
+        runner-type: "github"
+        benchmark: "fio"
+    - name: run get-repo-name
+      id: get-repo-name
+      uses: ./crucible-ci/.github/actions/get-repo-name
+
+  display-params:
+    runs-on: ubuntu-latest
+    needs: gen-params
+    steps:
+    - name: Echo gen-params outputs
+      run: echo "${{ toJSON(needs.gen-params.outputs) }}"
+
+  github-runners:
+    runs-on: ubuntu-latest
+    needs:
+    - gen-params
+    - display-params
+    strategy:
+      fail-fast: false
+      matrix:
+        scenario: ${{ fromJSON(needs.gen-params.outputs.github_hosted_scenarios) }}
+        userenv: ${{ fromJSON(needs.gen-params.outputs.userenvs) }}
+    steps:
+    - name: Matrix Parameters => (${{ matrix.scenario.enabled }}, ${{ matrix.scenario.endpoint }}, ${{ matrix.scenario.benchmark }}, ${{ matrix.userenv }})
+      run: |
+        echo "enabled=${{ matrix.scenario.enabled }}"
+        echo "endpoint=${{ matrix.scenario.endpoint }}"
+        echo "benchmark=${{ matrix.scenario.benchmark }}"
+        echo "userenv=${{ matrix.userenv }}"
+
+    - name: checkout crucible-ci default
+      if: ${{ inputs.crucible_ci_test != 'yes' }}
+      uses: actions/checkout@v3
+      with:
+        repository: perftool-incubator/crucible-ci
+        ref: main
+        path: crucible-ci
+    - name: checkout crucible-ci crucible_ci_test
+      if: ${{ inputs.crucible_ci_test == 'yes' }}
+      uses: actions/checkout@v3
+      with:
+        repository: perftool-incubator/crucible-ci
+        ref: ${{ inputs.crucible_ci_test_branch }}
+        path: crucible-ci
```

Comment:
I can't sleep well with this. Perhaps we can create a reusable checkout  step that replaces the logic here. These 2 ifs here are fairly ugly.

Reply:
Once everything is switched over to reusable workflows and all the logic is consolidated into the crucible-ci repo it should make cleaning up stuff a lot easier.

Another thing that will help is when we eventually get all the repos moved to `main` instead of `master`.  Right now we are having to track those branches on a per project basis since we have inconsistency...once everything is `main` it should simplify some things because we can then just assume `main` rather than having to pass the correct branch name around.

That said, we may not not ever be able to get rid of all of the ugliness here...there are limitations to the workflow syntax that we may not be able to overcome.